### ISSUE TITLE
Issue with slack OAuth behaviour when signed into non target workspace in browser - invalid_team_for_non_distributed_app

### ISSUE THREAD

**Issue by benmort** [outline/outline#1621]
**To Reproduce**
Steps to reproduce the behaviour:
1. Setup slack OAuth credentials for Workspace "A" on outline instance
2. Sign into a single slack Workspace "B" via web browser
3. Go to your outline instance '/auth/slack'
4. See error "invalid_team_for_non_distributed_app"

**Expected behaviour**
If you are not signed into correct workspace you should have the option to choose which workspace you would like to sign-in through. Alternatively be forced to sign-in with the desired workspace.

**Description**
It's a pretty simple case that if you're not signed into any workspace via web browser then this flow works great. the issue i have is that when my users are attempting to log in they are using many different slack workspaces via the desktop app mostly. they are unaware of what workspaces if any are signed in via browser. so their experience is that is just doesn't work, even through the are signed in via desktop slack just fine.

Basically i see one solve.
> specify the "team id" in the slack OAuth flow which would force sign-in from the desired workspace 🎉 
thoughts?

**Screenshots**
![Screen Shot 2020-11-05 at 10 18 34 am](https://user-images.githubusercontent.com/617526/98178212-56db7600-1f50-11eb-90e3-e942adeeef31.png)

**Outline (please complete the following information):**
- Install: self hosted
- Version: 0.49.0

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: chrome
 - Version: 86.0.4240.183

**Comment by tommoor** [comment]
> specify the "team id" in the slack OAuth flow which would force sign-in from the desired workspace 

Is this something that's available with Slack's API's? You can also ask that folks sign-in to Slack _first_, before signing into Outline – you will then be presented with a choice of accounts by Slack.

**Comment by benmort** [comment]
@tommoor the issue is that the slack oauth app needs to be distributed!
now it is giving the prompt to select workspace instead of failing if not signed in.
adding an optional workspace or "team" id then will let you lock distributed oath apps to a single workspace

**Comment by fehrlich** [comment]
I run into the issue when i accidentally set up `slack.signing-secret` and `slack.verification-token`. Removing the `slack.verification-token` fixed the issue for me.
